chia-bls/src/public_key.rs

```diff
@@ -217,10 +211,13 @@ impl Streamable for PublicKey {
         Ok(())
     }
 
-    fn parse(input: &mut Cursor<&[u8]>) -> chia_traits::Result<Self> {
-        Ok(Self::from_bytes(
-            read_bytes(input, 48)?.try_into().unwrap(),
-        )?)
+    fn parse<const TRUSTED: bool>(input: &mut Cursor<&[u8]>) -> chia_traits::Result<Self> {
+        let input = read_bytes(input, 48)?.try_into().unwrap();
```

Comment:
Why `unwrap` instead of `?`?

Reply:
`read_bytes()` guarantees to read 48 bytes (or fail). The `try_into()` is guaranteed to succeed in turning the slice into `[u8; 48]`.

An alternative could be to make `read_bytes()` take a const generic as the number of bytes to read, and return an array directly.

Reply:
Maybe a new function `read_bytes_const`? Not necessary for now though.